DataScience/DataScience.py

```diff
@@ -176,56 +132,121 @@ class CheckpointedModel:
     def __init__(self, ts, container, dir):
         self.ts = ts
         self.container = container
-        self.model = CachedBlob(container, '{0}model'.format(dir))
+        # self.model = CachedBlob(container, '{0}model'.format(dir))
         self.trackback = CachedBlob(container, '{0}model.trackback'.format(dir))
         self.trackback_ids = [line.rstrip('\n') for line in open(self.trackback.filename)]
 
+def get_single_checkpoint_model(item_tuple):
+    return CheckpointedModel(item_tuple[0], item_tuple[1], item_tuple[2])
+
 def get_checkpoint_models():
-    for date_container in block_blob_service.list_blobs('onlinetrainer', delimiter = '/'):
-        for time_container in block_blob_service.list_blobs('onlinetrainer', prefix = date_container.name, delimiter = '/'):
+    for current_date in dates_in_range(start_date, end_date):
+        for time_container in block_blob_service.list_blobs('onlinetrainer', prefix = current_date.strftime('%Y%m%d/'), delimiter = '/'):
             m = re.search('^([0-9]{4})([0-9]{2})([0-9]{2})/([0-9]{2})([0-9]{2})([0-9]{2})', time_container.name)
             if m:
-                ts = datetime(int(m.group(1)), int(m.group(2)), int(m.group(3)), int(m.group(4)), int(m.group(5)), int(m.group(6)))  
-                if ts.date() < start_date or ts.date() > end_date:  
-                    continue  
-                yield CheckpointedModel(ts, 'onlinetrainer', time_container.name)
+                ts = datetime(int(m.group(1)), int(m.group(2)), int(m.group(3)), int(m.group(4)), int(m.group(5)), int(m.group(6)))    
+                yield (ts, 'onlinetrainer', time_container.name)
 
-model_history = list(get_checkpoint_models())
-model_history.sort(key=lambda jd: jd.ts)
+if __name__ == '__main__':
```

Comment:
The parallelization is done in 
"with Pool(processes = 5) as p:
       data = p.map(download_data, joined)"